### ISSUE TITLE
Support lists in config UI

### ISSUE THREAD

**Issue by timothybrooks** [wandb/wandb#1240]
### Description

In the wandb web app, the overview page of a wandb run shows logged config fields. These commonly include hyperparameters such as learning rate or number of hidden layers. While the structure of config values is typically key-value pairs, it sometimes makes sense to use a list when the number of entries is variable, for example when a list of validation datasets may be provided to the run configuration.

The UI currently displays lists as a flattened text field, as the following example shows. This makes them harder to read. It would be great to support display of lists (and nested lists/dictionaries inside of them).

![Screenshot from 2020-09-16 15-55-22](https://user-images.githubusercontent.com/10535711/93401160-671b8280-f836-11ea-80aa-5faf9bf7e473.png)

**Comment by cvphelps** [comment]
Hi @timothybrooks, interesting request. For this example, could you describe a little more about how you want this to be rendered?

**Comment by timothybrooks** [comment]
I think ideally the config UI would support hierarchical drop down menus for nested dictionaries and lists. That way if I log many config params with the same prefix, (e.g. dataset.path, dataset.name, dataset.size) they can be neatly collapsed or expanded. Similarly, for lists it would be great to have the list elements live under a drop down menu. That way in my example above, I could click on val_datasets, a list of elements would show up underneath, and I could then further dig into the subfields of each element.

For an easier temporary fix, you could turn lists into dictionaries with keys being the element indices, so that val_datasets.0.path would refer to the path of the first list item.

**Comment by jhuebotter** [comment]
I second this request and would be interested if there are any updates, given that the stale label has been removed?

**Comment by kptkin** [comment]
This feature is implemented so closing as complete feel free to open if there is still missing features you would like to see.
